### PR TITLE
[Bugfix] Fix sticky navbar in safari

### PR DESCRIPTION
**Pull request by schultek**
### Checklist
- [x] I tested *all* changes and *all* related features with different users (student, tutor, instructor, admin) on the test server https://artemistest.ase.in.tum.de.
- [x] Client: I added multiple screenshots/screencasts of my UI changes

### Motivation and Context

Fix sticky navbar for safari
Fix scrolling to quiz questions in all browsers

### Description

Move sticky class to component. 
Take exam navigation bar  into account when calculating scroll position for quiz questions.
Tested on Safari, Firefox and Chrome.

### Steps for Testing

1. Log in to Artemis
2. Participate in Exam with quiz exercise with many questions
3. Scroll down
4. Use the bar on the left to navigate to quiz questions. Expected outcome: question is on the top of the page when using bar to navigate to quiz questions

### Screenshots

Bildschirmfoto 2020-09-22 um 17 16 32

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -92,7 +92,7 @@ springBoot {
 }
 
 tasks.withType(JavaCompile).each {
-    it.options.compilerArgs.add('--enable-preview')
+//    it.options.compilerArgs.add('--enable-preview')
```
Comment (philippbauch) [inline build.gradle:95]: Should this be part of the PR or is it just to avoid the failing build task?

**Review by philippbauch — APPROVED** [review]
Works fine on my machine, tested on Chrome 85 and Safari 13.1.

File: build.gradle (hunk 1 above)
Comment (TobiasPr) [reply build.gradle:95]: just to avoid the failing build task

Comment (TobiasPr) [reply build.gradle]: I just removed it, because Stephan Krusche provided a solution for it

**Comment by anditurdiu** [comment]
Does this PR also adress the need to scroll "less" in order to account for the height of the sticky bar?

**Comment by TobiasPr** [comment]
> Does this PR also adress the need to scroll "less" in order to account for the height of the sticky bar?

yes, it takes into account the exam-navigation-bar when scrolling in quiz exercises using the quiz sidebar

**Review by TobiasPr — APPROVED** [review]

**Review by jpbernius — APPROVED** [review]
LGTM :shipit: 🍎

**Pull request merged**